manifests/init.pp

```diff
@@ -38,8 +38,17 @@
       class { '::puppet_agent::windows::install': }
     }
     else {
-      class { '::puppet_agent::prepare': } ->
-      class { '::puppet_agent::install': } ->
+
+      if $::operatingsystem == 'SLES' and $::operatingsystemmajrelease == '10' {
+        $_package_file_name = "${puppet_agent::package_name}-${puppet_agent::params::master_agent_version}-1.sles10.${::architecture}.rpm"
```

Comment:
@highb I looked into this, but that name is dependent on puppet_agent::package_name which, given that it is a top level param could theoretically change, though I don't know why it would.  Given the current structure seems best to keep it as is.